lib/Runtime/Language/InlineCache.h

```diff
@@ -374,40 +373,33 @@ namespace Js
     public:
         static const bool IsPolymorphic = true;
 
-    private:
+    protected:
         FieldNoBarrier(InlineCache *) inlineCaches;
-        Field(FunctionBody *) functionBody;
         Field(uint16) size;
         Field(bool) ignoreForEquivalentObjTypeSpec;
         Field(bool) cloneForJitTimeUse;
 
         Field(int32) inlineCachesFillInfo;
 
-        // DList chaining all polymorphic inline caches of a FunctionBody together.
-        // Since PolymorphicInlineCache is a leaf object, these references do not keep
-        // the polymorphic inline caches alive. When a PolymorphicInlineCache is finalized,
-        // it removes itself from the list and deletes its inline cache array.
-        Field(PolymorphicInlineCache *) next;
-        Field(PolymorphicInlineCache *) prev;
-
-        PolymorphicInlineCache(InlineCache * inlineCaches, uint16 size, FunctionBody * functionBody)
-            : inlineCaches(inlineCaches), functionBody(functionBody), size(size), ignoreForEquivalentObjTypeSpec(false), cloneForJitTimeUse(true), inlineCachesFillInfo(0), next(nullptr), prev(nullptr)
+        PolymorphicInlineCache(InlineCache * inlineCaches, uint16 size)
+            : inlineCaches(inlineCaches), size(size), ignoreForEquivalentObjTypeSpec(false), cloneForJitTimeUse(true), inlineCachesFillInfo(0)
         {
-            Assert((size == 0 && inlineCaches == nullptr) ||
-                (inlineCaches != nullptr && size >= MinPolymorphicInlineCacheSize && size <= MaxPolymorphicInlineCacheSize));
         }
 
     public:
-        static PolymorphicInlineCache * New(uint16 size, FunctionBody * functionBody);
 
-        static uint16 GetInitialSize() { return MinPolymorphicInlineCacheSize; }
         bool CanAllocateBigger() { return GetSize() < MaxPolymorphicInlineCacheSize; }
         static uint16 GetNextSize(uint16 currentSize)
         {
             if (currentSize == MaxPolymorphicInlineCacheSize)
             {
                 return 0;
             }
+            else if (currentSize == MinPropertyStringInlineCacheSize)
+            {
+                CompileAssert(MinPropertyStringInlineCacheSize < MinPolymorphicInlineCacheSize);
+                return MinPolymorphicInlineCacheSize;
```

Comment:
Oh right, I misread MinPolymorphicInlineCacheSize and MinPropertyStringInlineCacheSize as the same

---
In reply to: [115339150](https://github.com/Microsoft/ChakraCore/pull/2883#discussion_r115339150) [](ancestors = 115339150)